src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -918,6 +921,26 @@ public String getFilter() {
     public void setHeaders(Map<String, Object> requestHeadersParam) {
         properties.put(ReplicationField.REQUEST_HEADERS, requestHeadersParam);
         replicationInternal.setHeaders(requestHeadersParam);
+        // Fix - https://github.com/couchbase/couchbase-lite-java-core/issues/1617
+        storeCookiesIntoCookieJar(requestHeadersParam);
+    }
+
+    private void storeCookiesIntoCookieJar(Map<String, Object> requestHeadersParam) {
+        try {
+            if (requestHeadersParam.containsKey("Cookie") && requestHeadersParam.get("Cookie") instanceof String) {
+                String cookieString = (String) requestHeadersParam.get("Cookie");
+                if (remote != null) {
+                    // NOTE: In case that the path of URL is not end with `/`,
+                    //       last segment of a path is not stored as a path of Cookie.
+                    Cookie cookie = Cookie.parse(HttpUrl.get(remote), cookieString);
+                    if (cookie != null && replicationInternal != null) {
```

Comment:
### for comment 1
- replicationInternal should not be null. So I will remove this.

### for comment 2
- Cookie's in Cookie jar always overwrite if it exists. However, passed cookie value in the headers does not includes expirers, domain, and path. So it could send two cookies in the final request header. As you advise, I feel it is better to remove from header.